### DISCUSSION TITLE
Bug Report: Mutations returns wrong url

### DISCUSSION THREAD

**Discussion by jln13x** [Shopify/storefront-api-feedback#130] [Help] [answered]
Storefront GraphQL API - Version `2202-07`

I have assigned a custom domain to my shopify store (as primary) but the mutation `checkoutCreate` still returns `foo.myshopify.com` as `webUrl`. 
In comparsion the query `cart` works as expected and returns my custom domain as `checkoutUrl`

**Comment by vixdug** [comment, marked as answer]
Hi @JLN13X try initializing the API (ex. `yourcustomdomain.com/api/graphql`) with your custom domain vs. the .myshopify.com domain. However, the cart is the most feature rich way to get to web checkout and what we recommend to use moving forward.

**Reply by jln13x** [reply]
Ye that worked. Still I would like consistency between the `checkoutCreate` and `cart` query
